app/src/main/java/org/mozilla/reference/browser/library/history/ui/HistoryFragment.kt

```diff
@@ -77,34 +75,20 @@ class HistoryFragment @JvmOverloads constructor(
         return view
     }
 
-    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
-        val layout = when (historyViewModel.viewMode) {
-            ViewMode.Normal -> R.menu.history_menu
-            ViewMode.Editing -> R.menu.history_multi_select_menu
-        }
-        inflater.inflate(layout, menu)
-    }
-
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        when (item.itemId) {
-            android.R.id.home, R.id.close -> {
-                onBackPressed()
-                return true
-            }
-            R.id.delete -> {
-                historyViewModel.deleteMultipleHistoryItem(historyViewModel.selectedItems)
-                historyViewModel.viewMode = ViewMode.Normal
-                return true
-            }
-        }
-        return super.onOptionsItemSelected(item)
+    private fun openHistoryItem(item: HistoryItem) {
+        context?.openToBrowserAndLoad(
+            searchTermOrUrl = item.url,
+            newTab = false,
+            from = BrowserDirection.FromHistory,
+            private = false
+        )
     }
 
     private fun openHistoryItems(items: Set<HistoryItem>, private: Boolean) {
         items.forEach { historyItem ->
             context?.openToBrowserAndLoad(
                 searchTermOrUrl = historyItem.url,
-                newTab = items.size != 1, // Load in the current tab
```

Comment:
There is this one case where you want to open one selected item in a private tab. So `newTab` here becomes false, and the selected item will open in the same tab and not in a new tab. And this is where we are having a bug where a private tab was opening like a normal tab in the same tab.

But this can be addressed with a single change:
```
newTab = items.size != 1 || private, // Load in the current tab
```
This is a bit harder to understand now.

Also, this would always open a history item clicked from the list in the new tab when in private mode.

So, I am thinking it's a good idea to have a distinction between a single item clicked in the list (opens in the same tab) and multiple items opened from the toolbar (opens the items in multiple tabs) in the form of two functions.